test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_image_arguments.cpp

```diff
@@ -0,0 +1,432 @@
+        imageInfo.format = &formats;
+        imageInfo.width = 4;
+
+        BufferOwningPtr<char> imageValues, outputData;
+        MTdataHolder d(gRandomSeed);
+        generate_random_image_data(&imageInfo, imageValues, d);
+        generate_random_image_data(&imageInfo, outputData, d);
+
+        char *host_ptr = (char *)imageValues;
+
+        clMemWrapper src_image = create_image_1d(
+            context, CL_MEM_READ_WRITE | CL_MEM_COPY_HOST_PTR, &formats,
+            image_desc.image_width, 0, host_ptr, nullptr, &error);
+        test_error(error, "create_image_1d failed");
+
+        clMemWrapper dst_image = create_image_1d(
+            context, CL_MEM_READ_WRITE | CL_MEM_COPY_HOST_PTR, &formats,
+            image_desc.image_width, 0, host_ptr, nullptr, &error);
+        test_error(error, "create_image_2d failed");
+
+        error = create_single_kernel_helper(context, &program, &kernel, 1,
+                                            &sample_const_arg_kernel,
+                                            "sample_test");
+        test_error(error, "Creating kernel failed");
+
+        clSamplerWrapper sampler = clCreateSampler(
+            context, CL_FALSE, CL_ADDRESS_NONE, CL_FILTER_NEAREST, &error);
+        test_error(error, "Unable to create sampler");
+
+        error = clSetKernelArg(kernel, 0, sizeof(cl_mem), &src_image);
+        test_error(error, "Unable to set indexed kernel arguments");
+
+        error = clSetKernelArg(kernel, 1, sizeof(cl_sampler), &sampler);
+        test_error(error, "Unable to set indexed kernel arguments");
+
+        error = clSetKernelArg(kernel, 2, sizeof(cl_mem), &dst_image);
+        test_error(error, "Unable to set indexed kernel arguments");
+
+        cl_ndrange_kernel_command_properties_khr props[] = {
+            CL_MUTABLE_DISPATCH_UPDATABLE_FIELDS_KHR,
+            CL_MUTABLE_DISPATCH_ARGUMENTS_KHR, 0
+        };
+
+        size_t globalDim[3] = { 4, 1, 1 }, localDim[3] = { 1, 1, 1 };
+
+        error = clCommandNDRangeKernelKHR(
+            command_buffer, nullptr, props, kernel, 1, nullptr, globalDim,
+            localDim, 0, nullptr, nullptr, &command);
+        test_error(error, "clCommandNDRangeKernelKHR failed");
+
+        error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+
+        error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 0,
+                                          nullptr, nullptr);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        error = clFinish(queue);
+        test_error(error, "clFinish failed.");
+
+        clMemWrapper new_image = create_image_1d(
+            context, CL_MEM_READ_WRITE | CL_MEM_COPY_HOST_PTR, &formats,
+            image_desc.image_width, 0, host_ptr, nullptr, &error);
+        test_error(error, "create_image_2d failed");
+
+        cl_mutable_dispatch_arg_khr arg_0{ 0, sizeof(cl_mem), &new_image };
```

Comment:
Should we be setting the arg with index `2` here rather than `0`? With index `0` we're overwriting `src` image, but we don't verify `dst_image` has the correct data.